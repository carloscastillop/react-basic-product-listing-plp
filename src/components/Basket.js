import React from 'react';
import * as Constant from '../Constants';
import '../styles/App.scss';

class Basket extends React.Component {

    render() {
        const {total, qty} = this.props;
        return (
            <React.Fragment>
                <div className="basketContainer mt-auto py-3 fixed-bottom">
                    <div className="container">
                        <p
                            className="d-inline-block m-0"
                            data-testid="qtyCounter"
                        >
                            <i className="fas fa-shopping-bag fa-2x mr-1"></i>
                            <span>x{qty}</span> item{qty !== 1 && <span>s</span> }
                        </p>
                        <p
                            className="total h3 float-right m-0"
                            data-testid="totalSum"
                        >
                            Total: <span>{Constant.CURRENCY}<strong>{total}</strong></span>
                        </p>
                    </div>
                </div>
            </React.Fragment>
        )
    }
}

export default Basket;