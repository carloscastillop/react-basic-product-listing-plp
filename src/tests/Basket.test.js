import React from 'react';
import ReactDOM from 'react-dom';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

import Basket from '../components/Basket';

afterEach(cleanup);
/**
 * Basic render
 */
it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Basket/>, div);
});

/**
 * Render empty basket, with 0 values (A)
 */
it('renders when basket is empty', () => {
    const {qty, total} = {qty: 0, total: 0};
    const {getByTestId} = render(<Basket total={total} qty={qty}/>)
    expect(getByTestId('qtyCounter')).toHaveTextContent('x0 items');
    expect(getByTestId('totalSum')).toHaveTextContent('Total: £0');
});

/**
 * Render basket, when items === 1 (B)
 */
it('Basket with items === 1', () => {
    const {qty, total} = {qty: 1, total: 10};
    const {getByTestId} = render(<Basket total={total} qty={qty}/>)
    expect(getByTestId('qtyCounter')).toHaveTextContent('x1 item');
    expect(getByTestId('totalSum')).toHaveTextContent('Total: £10');
});

/**
 * Render basket, when items > 1 (C)
 */
it('Basket with items > 1', () => {
    const {qty, total} = {qty: 10, total: 100};
    const {getByTestId} = render(<Basket total={total} qty={qty}/>)
    expect(getByTestId('qtyCounter')).toHaveTextContent('x10 items');
    expect(getByTestId('totalSum')).toHaveTextContent('Total: £100');
});

/**
 * Snapshot (A)
 */
it('matches snapshot A', () => {
    const {qty, total} = {qty: 0, total: 0};
    const tree = renderer.create(<Basket total={total} qty={qty}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

/**
 * Snapshot (B)
 */
it('matches snapshot B', () => {
    const {qty, total} = {qty: 1, total: 10};
    const tree = renderer.create(<Basket total={total} qty={qty}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

/**
 * Snapshot (C)
 */
it('matches snapshot C', () => {
    const {qty, total} = {qty: 10, total: 100};
    const tree = renderer.create(<Basket total={total} qty={qty}/>).toJSON();
    expect(tree).toMatchSnapshot();
});