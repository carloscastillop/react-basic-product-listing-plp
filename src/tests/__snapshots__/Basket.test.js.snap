// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot A 1`] = `
<div
  className="basketContainer mt-auto py-3 fixed-bottom"
>
  <div
    className="container"
  >
    <p
      className="d-inline-block m-0"
      data-testid="qtyCounter"
    >
      <i
        className="fas fa-shopping-bag fa-2x mr-1"
      />
      <span>
        x
        0
      </span>
       item
      <span>
        s
      </span>
    </p>
    <p
      className="total h3 float-right m-0"
      data-testid="totalSum"
    >
      Total: 
      <span>
        £
        <strong>
          0
        </strong>
      </span>
    </p>
  </div>
</div>
`;

exports[`matches snapshot B 1`] = `
<div
  className="basketContainer mt-auto py-3 fixed-bottom"
>
  <div
    className="container"
  >
    <p
      className="d-inline-block m-0"
      data-testid="qtyCounter"
    >
      <i
        className="fas fa-shopping-bag fa-2x mr-1"
      />
      <span>
        x
        1
      </span>
       item
    </p>
    <p
      className="total h3 float-right m-0"
      data-testid="totalSum"
    >
      Total: 
      <span>
        £
        <strong>
          10
        </strong>
      </span>
    </p>
  </div>
</div>
`;

exports[`matches snapshot C 1`] = `
<div
  className="basketContainer mt-auto py-3 fixed-bottom"
>
  <div
    className="container"
  >
    <p
      className="d-inline-block m-0"
      data-testid="qtyCounter"
    >
      <i
        className="fas fa-shopping-bag fa-2x mr-1"
      />
      <span>
        x
        10
      </span>
       item
      <span>
        s
      </span>
    </p>
    <p
      className="total h3 float-right m-0"
      data-testid="totalSum"
    >
      Total: 
      <span>
        £
        <strong>
          100
        </strong>
      </span>
    </p>
  </div>
</div>
`;
